shared/util.go

```diff
@@ -800,6 +800,11 @@ func IsUserConfig(key string) bool {
 	return strings.HasPrefix(key, "user.")
 }
 
+// IsProjectStorageConfig returns true if the key starts with the prefix "storage.project.".
+func IsProjectStorageConfig(key string) bool {
```

Comment:
Couple of points here:

- Code in the `shared` package is (originally) intended for consumption by both client and server, so if thats not required here then it shouldn't be in shared.
- Not really relevant here, but code in `shared` is licensed out as under Apache2 so that clients can consume these functions in their projects.
- Finally, we're trying to reduce and move away from using wide reaching and general "util" type files/packages and instead put functions into more concisely named packages, so I would like to avoid us adding new functions to `shared/util.go` as its about as general as it can get :)